db/dcrpg/tables.go

```diff
@@ -74,34 +74,61 @@ type TableVersion struct {
 	major, minor, patch uint32
 }
 
+// CompatibilityAction defines the action to be taken once the current and the
+// required pg table versions are compared.
+type CompatibilityAction int8
+
+const (
+	Rebuild CompatibilityAction = iota
+	Upgrade
+	ReIndex
+	OK
+	Unknown
+)
+
 // TableVersionCompatible indicates if the table versions are compatible
 // (equal), and if not, what is the required action (rebuild, upgrade, or
 // reindex).
-func TableVersionCompatible(required, actual TableVersion) string {
+func TableVersionCompatible(required, actual TableVersion) CompatibilityAction {
 	switch {
 	case required.major != actual.major:
-		return "rebuild"
+		return Rebuild
 	case required.minor != actual.minor:
-		return "upgrade"
+		return Upgrade
 	case required.patch != actual.patch:
-		return "reindex"
+		return ReIndex
 	default:
-		return "ok"
+		return OK
 	}
 }
 
 func (s TableVersion) String() string {
 	return fmt.Sprintf("%d.%d.%d", s.major, s.minor, s.patch)
 }
 
+// CompatibilityAction default stringer
```

Comment:
will make linters mad if it does not start with `// String `, so maybe `// String implements Stringer for CompatibilityAction` (the `type Stringer interface` in the `fmt` package)/